content/articles/get-started-with-a-lexical-analyzer/index.md

```diff
@@ -0,0 +1,534 @@
+```bash
+sudo apt-get install byacc
+```
+
+- One can also install the Bison package to generate a parser in C or C++ from a BNF notation. One can do it by running the command below:
+```bash
+sudo apt-get install bison++
+```
+
+#### Windows Installation
+
+To install Flex and Bison for Yacc, do the following:
+- Head over to this [site](http://gnuwin32.sourceforge.net/packages.html). All GNU utilities such as Bison and Flex are found there. Download Bison Setup found on the same site or by clicking [here](http://gnuwin32.sourceforge.net/packages/bison.htm).
+- Install it by running it.
+- Download Flex setup by clicking [here](http://gnuwin32.sourceforge.net/packages/flex.htm). Install it after the downloading process is complete.
+
+Add the binaries to the PATH variable by doing the following:
+- Open the folder in which the program files for Bison and Flex are installed. It is by default found in the `C:\Program Files (x86)\GnuWin32` file.
+- Copy its location path
+- Search in the Windows taskbar for `Environment Variables` and select the 'Edit the System Environment Variables' option.
+- Select the 'Environment Variables' option.
+- In the 'Path' option, paste the location of the bin folder. The folder path to be pasted is `C:\Program Files (x86)\GnuWin32\bin`.
+
+![Path location](/engineering-education/get-started-with-a-lexical-analyzer/path.png "Path location")
+
+- Save it and close the windows.
+
+### Architecture of Lex files
+Lex files have a `.l` extension.
+However, before creating a Lex file, let us first know the parts of a Lex file.
+
+#### Lex file parts
+Lex files have three main parts. These parts are the following:
+- Definition
+- Rules
+- Subroutines
+
+**Definition**: It contains literal blocks, definitions, internal table declarations, start conditions, and translations.
+It can contain things like the 'include' section found in the C program. This statement states the things that the program could need.
+Definitions can be written inside the `%{` and `%}` signs.
+
+**Rules and actions**: This section carries patterns and _C_ code(actions). These patterns can be in ECMAScript RegEx or wildcard format. It specifies the tokens the program checks, and the C code states what happens when they are found in the Actions section.
+Actions can be found inside the curly brackets({ ... }).
+The actions or macros can be optional too. Hence, they can be left blank.
+
+**Subroutines**: C codes that state additional user-defined functions executed when the patterns are found or not found.
+They can also contain other functions that the program may execute on the run.
+These are optional.
+
+The three sections are divided by delimiters (`%%`) as shown below:
+```
+1. Definitions section:
+
+%{    contains C code, usually consisting of the libraries
+        #define external global declarations that one wants to be part of the code for y.tab.c
+%}
+
+The file can include optional macro definitions for regular expressions.
+
+%%
```

Comment:
This is not very clear. Please explain more about this.

Reply:
let me check into it

Reply:
@srishilesh, these were previously spoken of in the parts before. The parts under the definition, rules, and subroutines headers in the article. Once the reader reads them (they are a little bit in-depth) when they reach this part, they will remember the previously read through brief hints in the screenshot

Reply:
Please review and tell me if it is well